components/CustomLink.vue

```diff
@@ -0,0 +1,93 @@
+<template>
+  <Component
+    :is="type"
+    :href="href ? link : null"
+    :to="to ? to : null"
+    :target="href ? '_blank' : null"
+    class="link"
+    :class="button && 'button'"
+    role="link"
+    tabindex="0"
+  >
+    <span :class="!button && 'link__content'">
+      <slot />
+    </span>
+  </Component>
+</template>
+
+<script>
+export default {
+  props: {
+    to: {
+      type: String,
+      required: true,
+    },
+    href: {
+      type: String,
+      required: true,
+    },
+    button: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  computed: {
+    type() {
+      if (this.to) {
+        return 'nuxt-link'
+      } else if (this.href) {
+        return 'a'
```

Comment:
Just a small suggestion. Since JS types are weird I prefer being explicit so maybe it is worth to check it like this:
```js
if (this.to !== undefined) {
  // ...
} else if (this.href !== undefined) {
  // ...
} else {
  // ...
}
```

JS has a lot of `falsy` values so better be sure on this :)